util/io.go

```diff
@@ -17,13 +17,37 @@
 package util
 
 import (
+	"errors"
 	"fmt"
 	"io"
 	"os"
 	"path/filepath"
 	"strings"
+	"syscall"
 )
 
+func MoveFile(src, dst string) error {
+	err := os.Rename(src, dst)
+	var errno syscall.Errno
+	if err != nil && errors.As(err, &errno) && errno == syscall.EXDEV {
+		// os.Rename() failed because src and dst are on different filesystems
+		// Must manually copy and delete the original file
+		return manuallyMoveFile(src, dst)
+	}
+	return err
+}
+
+func manuallyMoveFile(src, dst string) error {
+	tmpDst := dst + ".tmp"
```

Comment:
Can `dst` be a directory?  If so, you might imagine two different threads trying to move two different files to `/tmp`, but the both get the same temporary, `/tmp.tmp`.

More generally, there's probably a `tmpfile()` style function somewhere that ensures two threads never create the same file.

Reply:
https://pkg.go.dev/os#CreateTemp

Reply:
Destination can't be a directory for `os.Rename`, so I've outlawed that case in our copy and delete function as well. But I've still taken the advice to use `os.CreateTmp` in d21751413

Reply:
Ah, ok. Ignore my question about destination being a directory, elsewhere.

Reply:
It's still worth testing that I've outlawed the case properly. Tested in 299869729